src/Storages/Kafka/StorageKafka.cpp

```diff
@@ -458,15 +471,14 @@ bool StorageKafka::streamToViews()
     // It will be cancelled on underlying layer (kafka buffer)
     std::atomic<bool> stub = {false};
     copyData(*in, *block_io.out, &stub);
-    for (auto & stream : streams)
-        stream->as<KafkaBlockInputStream>()->commit();
 
-    // Check whether the limits were applied during query execution
-    bool limits_applied = false;
-    const BlockStreamProfileInfo & info = in->getProfileInfo();
-    limits_applied = info.hasAppliedLimit();
+    bool no_stalled_streams = true;
+    for (auto & stream : streams) {
+        no_stalled_streams = no_stalled_streams && !stream->as<KafkaBlockInputStream>()->isStalled();
+        stream->as<KafkaBlockInputStream>()->commit();
+    }
 
-    return limits_applied;
+    return no_stalled_streams;
```

Comment:
maybe better to replace this with `stalled_stream`, otherwise it is like double no-no, here and where the return value is checked?